lib/temporal/testing/temporal_override.rb

```diff
@@ -95,6 +107,8 @@ def start_locally(workflow, schedule, *input, **args)
           execution, workflow_id, run_id, workflow.disabled_releases, metadata
         )
 
+        Temporal::ThreadLocalContext.set(context)
```

Comment:
Two tests are failing and I have a hunch it's related to this line. 
 - `./spec/unit/lib/temporal/testing/temporal_override_spec.rb:106 # Temporal::Testing::TemporalOverride when testing mode is local Workflow.execute_locally restores original context after finishing successfully`
 - `/spec/unit/lib/temporal/testing/temporal_override_spec.rb:117 # Temporal::Testing::TemporalOverride when testing mode is local Workflow.execute_locally restores original context after failing`
 
 They don't fail when run by themselves, just when the whole test suite is run. Not sure what the best way to fix is.
 
 These tests aren't in the Cadence library.